docs/guides/periods.md

```diff
@@ -0,0 +1,228 @@
+# Periodic events and transitions
+
+Intervals can be handled declaratively with statecharts. To learn more, see the section in our [introduction to statecharts](./introduction-to-state-machines-and-statecharts/index.md#delayed-transitions).
+
+## Periodic transitions
+
+Events or transitions can be taken automatically every period of time. This is represented in a state definition in the `every` property, which maps millisecond periods to their transitions:
+
+```js
+const lightPeriodsMachine = createMachine({
+  id: 'lightPeriod',
+  initial: 'green',
+  context: {
+    timeSpentInGreen: 0,
+    timeSpentInYellow: 0
+  },
+  states: {
+    green: {
+      every: {
+        // every 1 second, timeSpentInGreen will be increased by 1 second
+        1000: {
+          actions: assign({
+            timeSpentInGreen: (context) => context.timeSpentInGreen + 1
+          })
+        }
+      },
+      on: {
+        GO_TO_YELLOW: 'yellow'
+      }
+    },
+    yellow: {
+      every: {
+        // every 0.5 seconds, timeSpentInYellow will be increased by 0.5 seconds
+        500: {
+          actions: assign({
+            timeSpentInYellow: (context) => context.timeSpentInYellow + 0.5
+          })
+        }
+      }
+    }
+  }
+});
+```
+
+Periodic events and transitions can be specified in the same way that you specify them on the `on: ...` property. They can be explicit:
+
+```js
+// ...
+states: {
+  green: {
+    every: {
+      1000: { actions: doSomeAction }
+    }
+  }
+}
+// ...
+```
+
+Periodic events can also be conditional with regard to a single interval value:
+
+```js
+// ...
+states: {
+  green: {
+    every: {
+      1000: [
+        { actions: doSomething, cond: 'trafficIsLight' },
+        { actions: doAnotherThing },
+      ]
+    }
+  }
+}
+// ...
+```
+
+### Periodic expressions on transitions <Badge text="4.4+" />
+
+Periodic events specified on the `every: { ... }` property can have dynamic priods, specified either by a string interval reference:
+
+```js
+const lightDelayMachine = createMachine(
+  {
+    id: 'lightDelay',
+    initial: 'green',
+    context: {
+      trafficLevel: 'low'
+    },
+    states: {
+      green: {
+        every: {
+          // after 1 second, transition to yellow
+          LIGHT_DELAY: { actions: doSomething }
+        }
+      },
+      yellow: {
+        after: {
+          YELLOW_LIGHT_DELAY: { actions: doTheYellowThing }
+        }
+      }
+      // ...
+    }
+  },
+  {
+    // String delays configured here
+    intervals: {
```

Comment:
It feels to me that in general, the difference between `delays` and `intervals` is so small that we could just merge them and keep using `delays` for intervals.